datafusion/optimizer/src/eliminate_filter.rs

```diff
@@ -88,7 +88,7 @@ mod tests {
 
     use crate::test::*;
 
-    fn assert_optimized_plan_equal(plan: &LogicalPlan, expected: &str) -> Result<()> {
+    fn assert_optimized_plan_equal(plan: LogicalPlan, expected: &str) -> Result<()> {
```

Comment:
A large number of test changes below are due to taking an owned LogicalPlan (to avoid having to copy it)